### PR TITLE
ORV2-2481 - PayBC financial transaction data fix

### PR DESCRIPTION
**Pull request by praju-aot**
Group the transactions by GLCode, sum the amount and set it to revenue field

# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Group transactions by Permit type GL Codes, sum the transaction amounts and forward to PayBC in the revenue field

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] npm run test
- [x] npm run test:e2e
- [x] manual testing via onRouteBc application 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1459-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1459-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1459-dops.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1459-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: vehicles/src/modules/permit-application-payment/payment/payment.service.ts [hunk 1]
```diff
@@ -91,10 +97,33 @@ export class PaymentService {
...
+    );
+
+    const groupedGlCodes = glCodeDetails.reduce((acc, { glCode, amount }) => {
+      acc.set(glCode, (acc.get(glCode) || 0) + amount);
+      return acc;
+    }, new Map<string, number>());
+
+    // Format the output string
+    const revenue = Array.from(groupedGlCodes.entries())
+      .map(([glCode, amount], index) => `${index + 1}:${glCode}:${amount}`)
```
Comment (krishnan-aot) [inline vehicles/src/modules/permit-application-payment/payment/payment.service.ts:124]: Why is it `index + 1`? I think an example in the code comment will make it easier to read.

File: vehicles/src/modules/permit-application-payment/payment/payment.service.ts [hunk 2]
```diff
@@ -91,10 +97,33 @@ export class PaymentService {
...
+      .join(', ');
```
Comment (krishnan-aot) [inline vehicles/src/modules/permit-application-payment/payment/payment.service.ts]: Is it comma separated or pipe?

File: vehicles/src/modules/permit-application-payment/payment/payment.service.ts (hunk 1 above)
Comment (praju-aot) [reply vehicles/src/modules/permit-application-payment/payment/payment.service.ts:124]: Done

File: vehicles/src/modules/permit-application-payment/payment/payment.service.ts (hunk 2 above)
Comment (praju-aot) [reply vehicles/src/modules/permit-application-payment/payment/payment.service.ts]: Pipe. Changed

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**